test/metabase/email/messages_test.clj

```diff
@@ -106,3 +109,68 @@
             emails (messages/render-pulse-email "America/Pacific" {} {} [part] nil)]
         (is (vector? emails))
         (is (map? (first emails)))))))
+
+(defn- get-positive-retry-metrics [retry]
+  (let [metrics (bean (.getMetrics retry))]
+    (into {}
+          (map (fn [field]
+                 (let [d (metrics field)]
+                   (when (pos? d)
+                     [field d]))))
+          [:numberOfFailedCallsWithRetryAttempt
+           :numberOfFailedCallsWithoutRetryAttempt
+           :numberOfSuccessfulCallsWithRetryAttempt
+           :numberOfSuccessfulCallsWithoutRetryAttempt])))
+
+(def test-email {:subject      "Test email subject"
+                 :recipients   ["test@test.com"]
+                 :message-type :html
+                 :message      "test mmail body"})
+
+(deftest send-email-retrying-test
+  (testing "send email succeeds w/o retry"
+    (let [test-retry (retry/random-exponential-backoff-retry "test-retry" (#'retry/retry-configuration))]
+      (with-redefs [email/send-email! mt/fake-inbox-email-fn
+                    retry/decorate    (fn [f]
+                                        (fn [& args]
+                                          (let [callable (reify Callable (call [_] (apply f args)))]
+                                            (.call (Retry/decorateCallable test-retry callable)))))]
```

Comment:
Can you pull this out as a `test-retry-decorate-fn` or something, similar to `fake-inbox-email-fn`? It's a bit verbose to repeat it in the setup for every test.